### PR TITLE
Change name of ALEApplication to MeshMovingApplication

### PR DESCRIPTION
**Pull request by AndreasWinterstein**
As proposed in the issue some weeks ago, I renamed the ALEApplication in MeshMoving Application, since it is used not only for FSI but also for other application cases.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (jcotela) [inline .vscode/settings.json]: This file should not be in github, please add it to your git exclude list.

File: applications/ExternalSolversApplication/externalsolvers_application.h [hunk 2]
```diff
@@ -6,8 +6,8 @@
 //
 //
 
-#if !defined(KRATOS_ALE_APPLICATION_H_INCLUDED)
-#define KRATOS_ALE_APPLICATION_H_INCLUDED
+#if !defined(KRATOS_MESH_MOVING_APPLICATION_H_INCLUDED)
+#define KRATOS_MESH_MOVING_APPLICATION_H_INCLUDED
```
Comment (jcotela) [inline applications/ExternalSolversApplication/externalsolvers_application.h]: I think you just found a big name error in the ExternalSolversApplication...

File: applications/MultiScaleApplication/CMakeLists.txt [hunk 3]
```diff
@@ -8,22 +8,22 @@ message("**** configuring KratosMultiScaleApplication ****")
 #
 ######################################################################################
 
-if(${MULTISCALE_APPLICATION_USE_EIGEN} MATCHES ON)
+if(${MULTISCMESH_MOVING_APPLICATION_USE_EIGEN} MATCHES ON)
```
Comment (jcotela) [inline applications/MultiScaleApplication/CMakeLists.txt]: Careful here! search and replace is not your friend.

File: applications/FSIapplication/python_scripts/partitioned_fsi_base_solver.py [hunk 4]
```diff
@@ -12,13 +12,13 @@
...
     "FSIApplication",
-    "ALEApplication",
+    "MeshMovingApplication",
     "FluidDynamicsApplication",
     "StructuralMechanicsApplication")
 
 # Import applications
 import KratosMultiphysics.FSIApplication as KratosFSI
-import KratosMultiphysics.ALEApplication as KratosALE
+import KratosMultiphysics.MeshMovingApplication as KratosALE
```
Comment (msandre) [inline applications/FSIapplication/python_scripts/partitioned_fsi_base_solver.py]: Shouldn't KratosALE also be renamed?

File: applications/FSIapplication/python_scripts/trilinos_partitioned_fsi_base_solver.py [hunk 5]
```diff
@@ -24,7 +24,7 @@
 import KratosMultiphysics.TrilinosApplication as KratosTrilinos
 import KratosMultiphysics.MappingApplication as KratosMapping
 import KratosMultiphysics.FSIApplication as KratosFSI
-import KratosMultiphysics.ALEApplication as KratosALE
+import KratosMultiphysics.MeshMovingApplication as KratosALE
```
Comment (msandre) [inline applications/FSIapplication/python_scripts/trilinos_partitioned_fsi_base_solver.py]: Shouldn't KratosALE also be renamed?

File: applications/FSIapplication/python_scripts/trilinos_partitioned_fsi_dirichlet_neumann_solver.py [hunk 6]
```diff
@@ -25,7 +25,7 @@
 import KratosMultiphysics.TrilinosApplication as KratosTrilinos
 import KratosMultiphysics.MappingApplication as KratosMapping
 import KratosMultiphysics.FSIApplication as KratosFSI
-import KratosMultiphysics.ALEApplication as KratosALE
+import KratosMultiphysics.MeshMovingApplication as KratosALE
```
Comment (msandre) [inline applications/FSIapplication/python_scripts/trilinos_partitioned_fsi_dirichlet_neumann_solver.py]: Shouldn't KratosALE also be renamed?

**Review by msandre — CHANGES_REQUESTED** [review]
I will approve as soon as the problem mentioned by @jcotela is fixed. Also, for the future please try to avoid committing files that are autoformatted for spaces only. There are many such files in this pr and it makes it difficult to track meaningful changes.

**Comment by armingeiser** [comment]
@msandre In the ```Files changed``` tab, on the top right, you can go to ```Diff settings``` and ```Hide whitespace changes```.

File: applications/FSIapplication/python_scripts/partitioned_fsi_base_solver.py (hunk 4 above)
Comment (rubenzorrilla) [reply applications/FSIapplication/python_scripts/partitioned_fsi_base_solver.py]: IMO should be. Maybe KratosMeshMoving?

File: applications/FSIapplication/python_scripts/trilinos_partitioned_fsi_base_solver.py (hunk 5 above)
Comment (rubenzorrilla) [reply applications/FSIapplication/python_scripts/trilinos_partitioned_fsi_base_solver.py]: Same

File: applications/FSIapplication/python_scripts/trilinos_partitioned_fsi_dirichlet_neumann_solver.py (hunk 6 above)
Comment (rubenzorrilla) [reply applications/FSIapplication/python_scripts/trilinos_partitioned_fsi_dirichlet_neumann_solver.py]: Same

**Comment by rubenzorrilla** [comment]
As a comment aside, I'm implementing a new explicit mesh moving technique (based on the FM-ALE method) for the embedded nodes initialization. Do you think it's better to merge it right now (it's implementation is finished but it has not been tested yet) or to merge it after this PR? I'm afraid of the conflicts that this PR may rise up.

**Comment by msandre** [comment]
@armingeiser that helped thanks. Still I would ask to avoid such commits in the future since it can still be a problem when one needs to go back through the git commit history to locate a bug. Probably git has a special command to hide white space changes, but I don't know if this is easy to do.

**Comment by AndreasWinterstein** [comment]
@msandre This happens in this case since I was using the replace function from vs code. I do not know why this happens also in files I did not touch.

**Comment by AndreasWinterstein** [comment]
@rubenzorrilla I do not know if this will cause too many conflicts. I also did not change anything concerning the names of the mesh solvers themselves and it worked.

**Comment by armingeiser** [comment]
@AndreasWinterstein VSCode automatically removes trailing whitespaces from lines when you save a file.
So even if you don`t actively change code, but hit Ctrl+S on a file, you will get some whitespace changes.

File: applications/ExternalSolversApplication/externalsolvers_application.h [hunk 7]
```diff
@@ -100,7 +100,7 @@ class KratosExternalSolversApplication : public KratosApplication {
 
     ///// Print object's data.
     void PrintData(std::ostream& rOStream) const override {
-        KRATOS_WATCH("in KratosALEApplication application");
+        KRATOS_WATCH("in KratosMeshMovingApplication application");
```
Comment (msandre) [inline applications/ExternalSolversApplication/externalsolvers_application.h]: This is wrong.

File: applications/ExternalSolversApplication/externalsolvers_application.h [hunk 8]
```diff
@@ -203,4 +203,4 @@ class KratosExternalSolversApplication : public KratosApplication {
 
 }  // namespace Kratos.
 
-#endif  // KRATOS_ALE_APPLICATION_H_INCLUDED  defined
+#endif  // KRATOS_MESH_MOVING_APPLICATION_H_INCLUDED  defined
```
Comment (msandre) [inline applications/ExternalSolversApplication/externalsolvers_application.h]: Wrong comment.

**Review by msandre — CHANGES_REQUESTED** [review]
Two small changes are still needed for external solvers app.

**Comment by AndreasWinterstein** [comment]
@msandre: if it is ok for you now, you can approve the PR. @dbaumgaertner has added the changes for the ShapeOptimizationApplication.

**Review by msandre — APPROVED** [review]
Thanks.

**Pull request merged**